### DISCUSSION TITLE
User and session is always null in Android (Kotlin) when using Twitter login

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#27259] [Questions]
Hello all,

So, I'm creating an Android application with Kotlin and Jetpack Compose. I want to integrate it with a Twitter login. I followed this [guide](https://supabase.com/docs/guides/auth/social-login/auth-twitter?language=kotlin&queryGroups=language&queryGroups=environment). I'm able to log in successfully, and the user is created in the Users table in the supabase server. However, the method always returns null when I try to get the current user `supabase.auth.currentUserOrNull()`. Any help would be appreciated. 

This is how I log in: 
```
Button(onClick = {
            scope.launch {
                try {
                    supabase.auth.signInWith(Twitter)
                } catch (e: Exception) {
                    Log.e("TwitterLoginScreen", e.message, e)
                }
            }
        }) {
            Text(text = "Login with Twitter")
        }
```
 This is the intent filter in my Activity: 
```
<intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="https"/>
                <data android:host="supabase_host"/>
                <data android:pathPrefix="/auth/v1/callback"/>
            </intent-filter>
```
And this is my client configuration
           
```
 object SupabaseClientProvider {
    val supabaseClient = createSupabaseClient(
        supabaseUrl = "URL",
        supabaseKey = "KEY"
    ) {
        install(Postgrest)
        install(Auth)
        install(Storage)
    }
}
```

**Comment by yuzefovichalex** [comment]
If you are trying to get the user when the app starts, the Auth plugin is most likely not initialized yet. You can call the `awaitInitialization()` method to make sure that the session is loaded from storage:
```kotlin
supabaseClient.auth.awaitInitialization()
val user = supabase.auth.currentUserOrNull()
```

**Reply by bcastello** [reply]
I was having a similar issue but it was with a regular login. This helped resolve my issue. Thanks.

**Reply by umardev500** [reply]
> If you are trying to get the user when the app starts, the Auth plugin is most likely not initialized yet. You can call the `awaitInitialization()` method to make sure that the session is loaded from storage:
> 
> ```kotlin
> supabaseClient.auth.awaitInitialization()
> val user = supabase.auth.currentUserOrNull()
> ```


thanks bro
